tests/JsonObjectSerializerTest.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Serializer\Tests;
+
+use PHPUnit\Framework\TestCase;
+use Yiisoft\Serializer\Tests\TestAsset\DummyData;
+use Yiisoft\Serializer\JsonObjectSerializer;
+
+final class JsonObjectSerializerTest extends TestCase
+{
+    private JsonObjectSerializer $serializer;
+
+    public function setUp(): void
+    {
+        $this->serializer = new JsonObjectSerializer();
+    }
+
+    public function testSerializeAndUnserialize(): void
+    {
+        $object = new DummyData('foo', 99, 1.1, [1, 'bar' => 'baz'], false);
+        $json = json_encode([
+            'string' => $object->getString(),
+            'int' => $object->getInt(),
+            'float' => $object->getFloat(),
+            'array' => $object->getArray(),
+            'bool' => $object->isBool(),
+        ]);
+
+        $this->assertSame($json, $this->serializer->serialize($object));
+        $this->assertEquals($object, $this->serializer->unserialize($json, DummyData::class));
+    }
+
+    public function testSerializeMultipleAndUnserializeMultiple(): void
+    {
+        $objects = [
+            $object1 = new DummyData('foo', 99, 1.1, [1], false),
+            $object2 = new DummyData('bar', 10, 2.2, [1, 2, 3], true),
+            $object3 = new DummyData('baz', 0, 3.3, ['foo' => 'bar'], false),
+        ];
+        $json = json_encode([
```

Comment:
Please use data providers and final json string instead

Reply:
Why? In this case, the array of objects is checked and the data provider is not needed.

Reply:
It's not interating anything. That's a single test data set so is OK.

Reply:
We also need extra cases:
1. Serialization of empty array
2. Serialization of one element in array
3. Serialization of object with recursive references

Also you didn't answer about using final json string instead of dynamic.

Reply:
json_encode result is predictable so I see no problem with using it.